pkg/deploy/service.go

```diff
@@ -193,11 +190,13 @@ func createDeploymentCanaryAnalysisStep(analysis *model.AnalysisStep) (*de.Analy
 	}, nil
 }
 
-func CreateAnalysisQueries(queries []model.Query, defaultAccount string) *[]de.AnalysisAnalysisQueries {
+func CreateAnalysisQueries(queries []model.Query, defaultAccount string) (*[]de.AnalysisAnalysisQueries, error) {
 	analysisQueries := make([]de.AnalysisAnalysisQueries, 0, len(queries))
 	for _, query := range queries {
-
 		if query.MetricProviderName == nil {
+			if defaultAccount == "" {
+				return nil, fmt.Errorf("analysis configuration block is present but default account not set")
```

Comment:
The error should probably say something like "metric provider must be provided either in the analysis config, as defaultMetricProvider, or in the query as metricProviderName".